### DISCUSSION TITLE
Borders of batch images showing in automatic mask generation in ArcGIS Pro

### DISCUSSION THREAD

**Discussion by SmithSamP** [opengeos/segment-geospatial#182] [Q&A]
Hello,

I have been using segment-geospatial to try and outline features on partially constructed subdivisions. I am using automatic mask segmentation in an ArcGIS Pro notebook. The only parameters that I have changed are the points_per_side kwarg, which I revised down due to my very high res imagery, along with enabling batch processing.

The problem I have encountered is that some binary images in the batch seem to have reversed values resulting in their outline being included as a mask. An small scale image of the masks can be seen below along with a larger scale image showing the resulting squares.
I have tried foreground=False, but all I got was a completely blank output.

Any help would be much appreciated.

![Example_binary_mask](https://github.com/opengeos/segment-geospatial/assets/92334211/b69cc857-67d7-473c-b1c1-9ad99abc3994)
![Example_Site_Picture](https://github.com/opengeos/segment-geospatial/assets/92334211/f1e557d2-54bb-4b6b-9115-33d9b22b64f7)

**Comment by giswqs** [comment]
It seems related to this issue https://github.com/opengeos/segment-geospatial/issues/7. I thought the issue has been resolved.

**Reply by SmithSamP** [reply]
Oh, ok. I hadn't seen that one. Thank you!

**Reply by SmithSamP** [reply]
Sorry to bring this back so late. But I installed the latest version from github and I am still getting the same issue with some of the batched images generating borders. I am running the model with foreground=True

**Reply by giswqs** [reply]
Did you try it with `foreground=False`?

**Reply by giswqs** [reply]
I see. You already tried it before. Can you provide complete source code that can reproduce the issue?

**Reply by SmithSamP** [reply]
[segment.txt](https://github.com/opengeos/segment-geospatial/files/12580551/segment.txt)

Uploading as a .txt as github won't allow .py. I have been running it in arcgis pro.

I have discovered that since upgrading the forground=False gets rid of the borders, but I was still hoping to use the foreground mask.

Thank you for the help. I have really been enjoying working with segment-geospatial.
